presto-hive/src/main/java/com/facebook/presto/hive/HivePageSource.java

```diff
@@ -230,22 +242,45 @@ public boolean isFinished()
     @Override
     public Page getNextPage()
     {
+        if (filterOnPrefilledValuesFailed) {
+            return null;
+        }
+
         try {
             Page dataPage = delegate.getNextPage();
             if (dataPage == null) {
                 return null;
             }
+
+            if (dataPage.getPositionCount() > 0 && filterOnPrefilledValuesException.isPresent()) {
+                if (filterOnPrefilledValuesException.get() instanceof PrestoException) {
+                    throw (PrestoException) filterOnPrefilledValuesException.get();
+                }
+                throw new RuntimeException(filterOnPrefilledValuesException.get());
+            }
+
             if (filterAndProjectPushedDown) {
                 if (!needsColumnMapping) {
                     return dataPage;
                 }
                 int batchSize = dataPage.getPositionCount();
-                List<Block> blocks = new ArrayList<>();
-                for (int fieldId = 0; fieldId < columnMappings.size(); fieldId++) {
+                int maxOutputChannel = -1;
+                for (int i = 0; i < outputChannels.length; i++) {
+                    if (outputChannels[i] == -1) {
```

Comment:
Seems unnecessary?  Doesn't change the behavior if you remove it.

Reply:
@tdcmeehan That's right. Will remove.